src/index.ts

```diff
@@ -207,25 +211,79 @@ export async function init(config: IClientConfig): Promise<IEppoClient> {
       throwOnFailedInitialization: true, // always use true here as underlying instance fetch is surrounded by try/catch
       skipInitialPoll: config.skipInitialRequest ?? false,
     };
+    EppoJSClient.instance.setConfigurationRequestParameters(requestConfiguration);
 
-    EppoJSClient.instance.setLogger(config.assignmentLogger);
+    // We have two at-bats for initialization: from the configuration store and from fetching
+    // We can resolve the initialization promise as soon as either one succeeds
+    let initFromConfigStoreError = undefined;
+    let initFromFetchError = undefined;
 
-    // default behavior is to use a LocalStorage-based assignment cache.
-    // this can be overridden after initialization.
-    EppoJSClient.instance.useCustomAssignmentCache(new LocalStorageAssignmentCache());
-    EppoJSClient.instance.setConfigurationRequestParameters(requestConfiguration);
-    await EppoJSClient.instance.fetchFlagConfigurations();
+    const attemptInitFromConfigStore = configurationStore
+      .init()
+      .then(() => {
+        if (!configurationStore.getKeys().length) {
+          // Consider empty configuration stores invalid
+          console.warn('Empty cached configuration');
+          initFromConfigStoreError = new Error('Configuration store was empty');
+          return '';
+        }
+        return 'config store';
+      })
+      .catch((e) => {
+        console.warn('Eppo SDK encountered an error initializing from the configuration store', e);
+        initFromConfigStoreError = e;
+      });
+    const attemptInitFromFetch = EppoJSClient.instance
+      .fetchFlagConfigurations()
+      .then(() => {
+        return 'fetch';
+      })
+      .catch((e) => {
+        console.warn('Eppo SDK encountered an error initializing from fetching', e);
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        initFromFetchError = e;
+      });
+
+    console.log('>>>>> ABOUT TO RACE!');
+
+    let initializationSource = await Promise.race([
+      attemptInitFromConfigStore,
+      attemptInitFromFetch,
+    ]);
+
+    console.log('>>>> loaded', configurationStore.getKeys().length, 'flags from', initializationSource, configurationStore.getKeys());
+
+    if (!initializationSource) {
```

Comment:
because of the `catch()` statements in the racing promises, no error will be _thrown_ upon race completion, but if the "winner" failed then no source will be set from that method, and we need to try the other. We can tell because we capture errors.